addons/mail/static/src/components/chatter/chatter.xml

```diff
@@ -38,6 +38,15 @@
                         />
                     </t>
                     <t t-if="chatter.threadView">
+                        <t t-if="chatter.threadView.nonEmptyMessages.length === 0 and (chatter.thread.activities.length > 0 or chatter.isAttachmentBoxVisible)">
```

Comment:
Please make a new computed field `isActivityBoxVisible` that uses `chatter.thread and chatter.hasActivities and chatter.thread.activities.length > 0` because current code is unreliable (condition to display the activitybox is different than the condition you added here)

Reply:
@seb-odoo we already have a field `isActivityBoxVisible` in the chatter which is not a computed field and clicking on the `o_ActivityBox_title` button will just toggle the value. and yes `hasActivities` field which determines the activity box available in the chatter. So both of these are not based on the length of activities. You mean to introduce a new one that uses `chatter.hasActivities and chatter.thread.activities.length > 0` right?

Because `isActivityBoxVisible` field behavior is just toggled the value while clicking on the title button (i.e. planned activities separator) It just folds/unfolds the activities.

Reply:
Yes sorry for the confusion with the name. The current `isActivityBoxVisible` is actually badly named, because the activity box component is shown even if it's value is `false`.
Current `isActivityBoxVisible` should be renamed to something like `isActivityTitleBadgeVisible` (based on the classname of the element that it renders visible in the DOM).

And then we can have a real `isActivityBoxVisible` that depends on `chatter.hasActivities and chatter.thread.activities.length > 0.`

Reply:
Note that `chatter.hasActivities` is also badly named, it describes if the activity feature should be enabled on the current chatter, not if there are actually actives existing or not. So I would suggested to rename it too.